src/easyscience/base_classes/new_base.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import annotations
+
+#  SPDX-FileCopyrightText: 2025 EasyScience contributors  <core@easyscience.software>
+#  SPDX-License-Identifier: BSD-3-Clause
+#  © 2021-2025 Contributors to the EasyScience project <https://github.com/easyScience/EasyScience
+from inspect import getfullargspec
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any
+    from typing import Dict
+    from typing import Iterable
+    from typing import List
+    from typing import Optional
+    from typing import Set
+
+from easyscience import global_object
+
+from ..global_object.undo_redo import property_stack
+from ..io.serializer_base import SerializerBase
+
+
+class NewBase:
+    """
+    This is the new base class for easyscience objects.
+    It provides serialization capabilities as well as unique naming and display naming.
+    """
+
+    def __init__(self, unique_name: Optional[str] = None, display_name: Optional[str] = None):
+        self._global_object = global_object
+        if unique_name is None:
+            unique_name = self._global_object.generate_unique_name(self.__class__.__name__)
+            self._default_unique_name = True
+        else:
+            self._default_unique_name = False
+        if not isinstance(unique_name, str):
+            raise TypeError('Unique name has to be a string.')
+        self._unique_name = unique_name
+        self._global_object.map.add_vertex(self, obj_type='created')
+        if display_name is not None and not isinstance(display_name, str):
+            raise TypeError('Display name must be a string or None')
+        self._display_name = display_name
+
+    @property
+    def _arg_spec(self) -> Set[str]:
+        """
+        This method is used by the serializer to determine which arguments are needed
+        by the constructor to deserialize the object.
+        """
+        spec = getfullargspec(self.__class__.__init__)
```

Comment:
It seems `inspect.signature()` is a modern/updated version of `getfullargspec`. The latter was supposed to be deprecated in >3.7 but was left for backward compatibility. We should move to modern/maintained calls.